### DISCUSSION TITLE
animation timing function

### DISCUSSION THREAD

**Discussion by MBSN91** [tailwindlabs/tailwindcss#9579] [Help] [answered]
Hi there,

I have a custom keyframe animation which works great, except when I try to add animation-timing-function. Tailwind throws an error which says:
Tailwind CSS: Unexpected token (65:21)

```
keyframes: {
        float: {
          '0%': {
            transform: 'translatey(0px)',

          },
          '25%':{
            transform: 'translatey(-75px)',
          },
          '50%':{
            transform: 'translatey(0px)',
            animation-timing-function: 'linear', <!-- this gives me an error, works flawlessly without. -->
          },  
          '75%':{
            transform: 'translatey(75px)',
          },
          '100%':{
            transform: 'translatey(0px)',
          },
        }
      },

```
I want to have different animation timing as it only needs ease-in-out at the beginning and the end of the animation, not in-between. Does tailwind not support this? Do I need to look for another solution outside of tailwind?

**Comment by wongjn** [comment, marked as answer]
Your error comes from the fact that you cannot have hyphens in an unquoted JavaScript object key, you could either convert the key to camel case:
```diff
   transform: 'translatey(0px)',
-  animation-timing-function: 'linear', <!-- this gives me an error, works flawlessly without. -->
+  animationTimingFunction: 'linear',
 },
```
Or quote the key:
```diff
   transform: 'translatey(0px)',
-  animation-timing-function: 'linear', <!-- this gives me an error, works flawlessly without. -->
+  'animation-timing-function': 'linear',
 },
```

**Reply by MBSN91** [reply]
Thank you! Now I know how to approach this. Works like a charm 💫
